src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -222,5 +227,31 @@ private void DisposeBuffers()
             ((IDisposable)CurrentRenderTree).Dispose();
             _latestDirectParametersSnapshot?.Dispose();
         }
+
+        internal Task DisposeInBatchAsync(RenderBatchBuilder batchBuilder)
```

Comment:
`ComponentState` is already `internal`

```suggestion
        public Task DisposeInBatchAsync(RenderBatchBuilder batchBuilder)
```